From darnok@68k.org Thu Jun 15 09:08:35 2006
Date: Thu, 15 Jun 2006 12:08:30 -0400
From: Konrad Rzeszutek <konradr@redhat.com>
To: Greg KH <greg@kroah.com>
Subject: PCI: fix memory leak in MMCONFIG error path
Message-ID: <20060615160830.GD3242@andromeda.dapyr.net>
Content-Disposition: inline

This a bit late (yours patch was posted about a year ago), but
a co-worker of spotted part of the code that looks like a memory
leak. Looking at the code it seems that pci_mmcfg_config should
be free-ed if MMCONFIG is above 4GB.


From: Konrad Rzeszutek <konradr@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/i386/kernel/acpi/boot.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/arch/i386/kernel/acpi/boot.c
+++ gregkh-2.6/arch/i386/kernel/acpi/boot.c
@@ -202,6 +202,8 @@ int __init acpi_parse_mcfg(unsigned long
 		if (mcfg->config[i].base_reserved) {
 			printk(KERN_ERR PREFIX
 			       "MMCONFIG not in low 4GB of memory\n");
+			kfree(pci_mmcfg_config);
+			pci_mmcfg_config_num = 0;
 			return -ENODEV;
 		}
 	}
